Field encryption curated sample (#37)
realm/realm-kotlin-samples: AppServicesUsageSamples/README.md

# App Services Usage Samples

Reference app that show cases different design patterns and examples of using the Realm Kotlin SDK with Atlas App Services.

## Samples

### [Property level encryption](apps/property-encryption/README.md)

This demo shows the process of protecting users' sensitive data by employing end-to-end encryption techniques while guaranteeing access from any user's device.

## Demo app structure

The project has been structured in two main folders:

- Demo - Android app containing the different samples. Samples have been separated in different packages.
- Apps - Atlas App services apps required by each sample.

## Getting started

The demos are indendepent of each other, this means that it is not required to install all the app services app samples to test an individual sample.

To begin, locate the App services app sources that you wish to install. We have conveniently linked them in the Samples list of this document.

> Next, follow the steps outlined in the [Atlas documentation](https://www.mongodb.com/docs/atlas/app-services/apps/create/) to setup the apps. These docs will guide you through the process and help troubleshoot any issue you might encounter.

Reviewer: I think this is a little sparse. At least I had troubles just getting up and running. Maybe we need to elaborate on the steps and show the actual import commands, or at at least point to some elaborate instructions, because simply creating an app and trying to upload the app didn't do the trick for me. Lets discuss offline.

# App Services Usage Samples

Reference app that show cases different design patterns and examples of using the Realm Kotlin SDK with Atlas App Services.

## Samples

### [Property level encryption](apps/property-encryption/README.md)

This demo shows the process of protecting users' sensitive data by employing end-to-end encryption techniques while guaranteeing access from any user's device.

## Demo app structure

The project has been structured in two main folders:

- `demo` - Android app containing the different samples. Samples have been separated in different packages.
- `apps` - Atlas App services apps required by each sample.

## Getting started

The demos are indendepent of each other, which means that it is not required to install all the app services app samples to test an individual sample.

To get started, locate the App services app sources in `apps` that you wish to install. We have conveniently linked them in the Samples list of this document.

> Next, follow the steps outlined in the [Atlas documentation](https://www.mongodb.com/docs/atlas/app-services/apps/create/) to setup an Atlas account and `realm-cli`, skip the app creation steps, then create and import the app with the following commands:

Reviewer: Better with the explicit import command, but still think it requires a bit of insights before the users can do it. 

I would rephrase it to very explicit steps. Something like: 
- Follow the https://www.mongodb.com/docs/atlas/app-services/apps/create/#before-you-begin steps
- Create a "Cluster0" 
- Login
- Import

If the "Cluster0" is not created the import will not work.

- `demo` - Android app containing the different samples. Samples have been separated in different packages.
- `apps` - Atlas App services apps required by each sample.

## Getting started

The demos are indendepent of each other, which means that it is not required to install all the app services app samples to test an individual sample.

To get started, locate the App services app sources in `apps` that you wish to install. We have conveniently linked them in the Samples list of this document.

Next, follow the "Before you begin" steps outlined in the [Atlas documentation](https://www.mongodb.com/docs/atlas/app-services/apps/create/) to setup an Atlas account and `realm-cli` (See the "App Services CLI" tab). Skip the app creation steps, but create and import the app with the following commands:

```bash
# If not created, create a cluster in Atlas. See https://www.mongodb.com/basics/clusters/mongodb-cluster-setup#creating-a-mongodb-cluster

# If not logged in, log in first
realm-cli login --api-key <your new public key> --private-api-key <your new private key>

# Move to the app directory
cd app/[APP-NAME]

# Import the current app directory
realm-cli import 
```

> The `realm-cli import` command would prompt for the app configuration details, the default parameters should be right for most cases. Then the tool would publish the app and return the app-id for the newly creted app. Example:

Reviewer 2: ```suggestion
The `realm-cli import` command will prompt for the app configuration details, the default parameters should be suitable for most cases. Then the tool will publish the app and return the app-id for the newly created app. Example:
```
